index.js

```diff
@@ -23,13 +23,34 @@ module.exports.pitch = function (request) {
 		"// Hot Module Replacement",
 		"if(module.hot) {",
 		"	// When the styles change, update the <style> tags",
-		"	if(!content.locals) {",
-		"		module.hot.accept(" + loaderUtils.stringifyRequest(this, "!!" + request) + ", function() {",
-		"			var newContent = require(" + loaderUtils.stringifyRequest(this, "!!" + request) + ");",
-		"			if(typeof newContent === 'string') newContent = [[module.id, newContent, '']];",
+		"	module.hot.accept(" + loaderUtils.stringifyRequest(this, "!!" + request) + ", function() {",
+		"		var newContent = require(" + loaderUtils.stringifyRequest(this, "!!" + request) + ");",
+		"		if(typeof newContent === 'string') newContent = [[module.id, newContent, '']];",
+		"		var localsSame = true;",
+		"		if(!content.locals && newContent.locals) {",
+		"			localsSame = false;",
+		"		} else if(content.locals && !newContent.locals) {",
+		"			localsSame = false;",
+		"		} else if(content.locals && newContent.locals) {",
+		"			var oldKeys = Object.keys(content.locals);",
+		"			var newKeys = Object.keys(newContent.locals);",
+		"			if(oldKeys.length !== newKeys.length) {",
```

Comment:
```js
const length = Object.keys(content.locals).length - Object.keys(newContent.locals).length

if (length > 0) {
   ...
}
```

Reply:
Why? I find this harder to understand. And I use `oldKeys` a couple of lines down.

Reply:
`const length` – the length of what? There’s nothing with length 0 around.

Reply:
Yep, sry I meant

```js
if (!length) {
  ...
}
```

Reply:
Fixed in the latest commit.